### PR TITLE
Replace bcftools/bgzip by tabix/bgzip

### PR DESCRIPTION
**Pull request by MaxUlysse**
cf #134 & #179

Decision was taken to rename bcftools/bgzip by tabix/bgzip and use the tabix container

### CODE REVIEW

**Comment by MaxUlysse** [comment]
Code is now more modular and can bgzip any file

File: software/tabix/bgzip/main.nf [hunk 1]
```diff
@@ -4,31 +4,31 @@ include { initOptions; saveFiles; getSoftwareName } from './functions'
...
-    tuple val(meta), path("*.gz"), emit: vcf
+    tuple val(meta), path("*.gz"), emit: file
     path  "*.version.txt"        , emit: version
 
     script:
     def software = getSoftwareName(task.process)
     def prefix   = options.suffix ? "${meta.id}${options.suffix}" : "${meta.id}"
     """
-    bgzip -c $options.args $vcf > ${prefix}.vcf.gz
+    bgzip -c $options.args $file > ${prefix}.${file.getExtension()}.gz
```
Comment (drpatelh) [inline software/tabix/bgzip/main.nf:31]: Nice 👍🏽

File: software/tabix/bgzip/main.nf [hunk 2]
```diff
@@ -4,31 +4,31 @@ include { initOptions; saveFiles; getSoftwareName } from './functions'
...
+        container "quay.io/biocontainers/tabix:0.2.6--ha92aebf_0"
     }
 
     input:
-    tuple val(meta), path(vcf)
+    tuple val(meta), path(file)
 
     output:
-    tuple val(meta), path("*.gz"), emit: vcf
+    tuple val(meta), path("*.gz"), emit: file
```
Comment (drpatelh) [inline software/tabix/bgzip/main.nf:24]: I always get paranoid about using variables that are native language variables.

File: tests/software/tabix/bgzip/main.nf [hunk 3]
```diff
@@ -2,13 +2,13 @@
 
 nextflow.enable.dsl = 2
 
-include { BCFTOOLS_BGZIP } from '../../../../software/bcftools/bgzip/main.nf' addParams( options: [:] )
+include { TABIX_BGZIP as BGZIP_VCF } from '../../../../software/tabix/bgzip/main.nf' addParams( options: [:] )
```
Comment (drpatelh) [inline tests/software/tabix/bgzip/main.nf:5]:
```suggestion
include { TABIX_BGZIP } from '../../../../software/tabix/bgzip/main.nf' addParams( options: [:] )
```
```suggestion
include { TABIX_BGZIP as BGZIP_VCF } from '../../../../software/tabix/bgzip/main.nf' addParams( options: [:] )
```

File: tests/software/tabix/bgzip/main.nf [hunk 4]
```diff
@@ -2,13 +2,13 @@
...
 
-workflow test_bcftools_bgzip {
+workflow test_tabix_bgzip_vcf {
 
     def input = []
     input = [ [ id:'test' ], // meta map
               [ file("${launchDir}/tests/data/vcf/test.vcf", checkIfExists: true) ]]
 
-    BCFTOOLS_BGZIP ( input )
+    BGZIP_VCF ( input )
```
Comment (drpatelh) [inline tests/software/tabix/bgzip/main.nf:13]:
```suggestion
    TABIX_BGZIP ( input )
```

File: software/tabix/bgzip/main.nf (hunk 2 above)
Comment (drpatelh) [reply software/tabix/bgzip/main.nf:24]: If it works...it works 🤷🏽

**Comment by drpatelh** [comment]
Appears I broke something here. Sorry! Hopefully, it is relatively easy to fix. Will approve because everything else looks ok. Feel free to merge when tests pass.

**Review by drpatelh — APPROVED** [review]

File: software/tabix/bgzip/main.nf (hunk 2 above)
Comment (MaxUlysse) [reply software/tabix/bgzip/main.nf:24]: good point, I'll change that for `gz`

**Comment by drpatelh** [comment]
👌🏽

**Pull request merged**